internal/store/store.go

```diff
@@ -21,8 +23,24 @@ func New(reader io.Reader, logger logrus.FieldLogger) (*Store, error) {
 		return nil, errors.Wrap(err, "failed to parse stream")
 	}
 
+	if err := validatePlugins(plugins); err != nil {
+		return nil, errors.Wrap(err, "failed to validate plugins")
+	}
+
 	return &Store{
 		plugins,
 		logger,
 	}, nil
 }
+
+func validatePlugins(plugins []*model.Plugin) error {
+	for _, plugin := range plugins {
+		if plugin.Manifest.Id == "" {
+			return errors.Errorf("plugin manifest Id is empty %+v", plugin)
+		}
+		if _, err := semver.Parse(plugin.Manifest.Version); err != nil {
+			return errors.Wrapf(err, "failed to parse manifest version for manifest.Id %s", plugin.Manifest.Id)
+		}
```

Comment:
Out of scope for this PR, but there should be a `Manifest.IsValid() (bool, err)` method in the server.

Reply:
Good idea, will create a jira for it shortly.

Reply:
@hanzei created a jira for this https://mattermost.atlassian.net/browse/MM-20168